src/tlo/methods/symptommanager.py

```diff
@@ -583,6 +602,16 @@ def clear_symptoms(self, person_id: Union[int, Sequence[int]], disease_module: M
         sy_columns = [self.get_column_name_for_symptom(sym) for sym in self.symptom_names]
         self.bsh.unset(person_id, disease_module.name, columns=sy_columns)
 
+        # Update bookkeeping
+        for pid in person_id:
+            for sym in self.symptom_names:
+                symptom_col = self.get_column_name_for_symptom(sym)
+                if self.bsh.is_empty(pid, columns=symptom_col):
+                    self.symptom_tracker[pid].discard(sym)
+            # Clean up if person has no symptoms left
+            if not self.symptom_tracker[pid]:
```

Comment:
as above, no real need to delete the key for the person when they have no symptoms...?